gst-plugin/src/drpai-models/base_post_processor.cpp

```diff
@@ -0,0 +1,115 @@
+//
+// Created by matin on 03/11/24.
+//
+
+#include "base_post_processor.h"
+#include <fstream>
+
+/// Renders bounding boxes on the image using the latest detections.
+/// This function usually gets overridden by the child class. But it is not a must.
+/// @param [in] img Reference to the image to be rendered on.
+void BasePostProcessor::render_detections_on_image(Image &img) {
+    std::unique_lock lock (mutex);
+    for (const auto& detection: last_det)
+    {
+        /* Draw the bounding box on the image */
+        img.draw_rect(detection.bbox, detection.to_string_hr());
+    }
+}
+
+/// Renders texts at the corner of the image using the list of corner texts.
+/// This function usually gets overridden by the child class. But it is not a must.
+/// @param [in] img Reference to the image to be rendered on.
```

Comment:
same here, what texts are we talking about and where do they get generated - i not overridden

Reply:
Right. I will move the text render into the `Image` class.